### ISSUE TITLE
Schema $ref values are not camelized when 'convert_casing' is enabled

### ISSUE THREAD

**Issue by takar** [pgjones/quart-schema#12]
When a schema is generated that contains a '$ref' to another object, the component definition is camelcased but values that reference this component are not.

For example, when defining the following objects:
```
@dataclass
class Employee:
    name: str

@dataclass
class Employees:
    resources: List[Employee]
```

The derived schema is:
```
{
  "components": {
    "schemas": {
      "employee": {  <-- camelcased
         ...
      }}},
  ...
  "paths": {
    "/employees/": {
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "resources": {
                      "items": {
                        "$ref": "#/components/schemas/Employee"  <-- NOT camelcased
   ...
```

**Comment by pgjones** [comment]
Can you provide a snippet that reproduces this, please?

**Comment by takar** [comment]
Here is a small example that demonstrates this issue:

```
from dataclasses import dataclass
from quart import Quart
from quart_schema import QuartSchema, validate_response

app = Quart(__name__)
QuartSchema(app, convert_casing=True)

@dataclass
class Bar:
    id: int

@dataclass
class Foo:
    bar: Bar

@app.get("/")
@validate_response(Foo)
async def index() -> Foo:
    return {'bar': {'id': 1}}

app.run()
```
If you run the app and go to localhost:5000/docs and open the `GET /` endpoint, it gives the following error:
```
Resolver error at paths./.get.responses.200.content.application/json.schema.properties.bar.$ref
```
When inspecting the JSON, the `Foo` response refers to a `Bar` definition (`"#/components/schemas/Bar"` - but there exists only a definition for `bar`.

Hope this clarifies!

**Comment by pgjones** [comment]
Should be fixed in 8dc493eae424c072a1817905c2635e8934fd4071
